kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -856,6 +865,10 @@ private void encryptPassword(Map<String, Object> propertiesToUpdate, String key,
     }
 
     private void encryptConfigs(List<? extends ComponentConfiguration> configs) {
+        if (configs == null) {
```

Comment:
XmlComponentConfigurations marshals the same with null and empty component configurations. Therefore it might be legal to store such a snapshot (although impossible in the current implementation). Meaning that this extra check is a safeguard from a wrong call that doesn't have to fail. No configs, no passwords, no change. And the same with null - no change.
If you still don't want the change, I'll remove it...